librad/src/keys/device.rs

```diff
@@ -137,6 +139,20 @@ impl Signature {
     pub fn verify(&self, data: &[u8], pk: &PublicKey) -> bool {
         sign::verify_detached(self, &data, pk)
     }
+
+    pub fn from_hex_string(s: &str) -> Result<Self, ()> {
```

Comment:
`Option` could work.
However in these cases I still prefer `Result` because for me the meaning of `None` is not really "error", it is just "nothing is here or has been found here", while in this case an actual error happened (the string was not a proper hex encoding of a signature).
Probably this would be a good place where to use `format_err!` from `failure`, if you agree I'll do it.

Reply:
I would say that `None` also signifies a failure in the computation, but we won't provide error information, or error information is implicit, e.g. divide by zero doesn't need an enumeration, it's one case so `None` would be fine.

`Restult<A, ()>` is not useful since there's no useful information. `()` is ambiguous in the case of an error, in face it's as useful as `None` and so `Result<A, ()>` is equivalent (some might say isomorphic) to `Option<A>` :)

So I'm open to what your final decision is, but my suggestions are either:
* Use an error type other than `()`, using `failure` could be an option but I'm actually not 100% on how this library works so it's your call :grin: 
* Use `Option`

I hope that's helpful :)

Reply:
I used `format_err!`.